File: app/src/main/java/org/scp/gymlog/model/Order.java
package org.scp.gymlog.model;

public enum Order {
    ALPHABETICALLY("alphabetically"),
    LAST_USED("last_used");

    public String name;
    Order(String name) {
        this.name = name;
    }

    public static Order getByName(String name) {
        for (Order order: Order.values()) {
            if (order.name.equals(name)){
                return order;
            }
        }
        throw new IllegalArgumentException("No order found for name \""+name+"\"");
    }
}

File: app/src/main/java/org/scp/gymlog/model/Order.kt
package org.scp.gymlog.model

import java.util.Arrays

enum class Order(val code: String) {
    ALPHABETICALLY("alphabetically"),
    LAST_USED("last_used");

    companion object {
        fun getByCode(code: String): Order {
            return Arrays.stream(values())
                    .filter { order: Order -> order.code == code }
                    .findFirst()
                    .orElseThrow { IllegalArgumentException("No order found for name \"$code\"") }
        }
    }
}